src/logger/src/logger.rs

```diff
@@ -244,6 +245,15 @@ impl Logger {
         self
     }
 
+    /// Get the current thread's name.
+    fn get_thread_name(&self) -> String {
+        let handle = thread::current();
+        match handle.name() {
+            Some(name) => name.to_string(),
+            None => "-".to_string(),
+        }
+    }
```

Comment:
You can use `&str` instead of `String` to avoid the overhead of building `String` objects if not needed:

```suggestion
    fn get_thread_name(&self) -> &str {
        let handle = thread::current();
        match handle.name() {
            Some(name) => name,
            None => "-",
        }
    }
```
 
Which can be shortened to:

```suggestion
    fn get_thread_name(&self) -> &str {
        thread::current().name().unwrap_or("-")
    }
```

Reply:
Thank you for the suggestion, we added this to the latest commit! 

However, wouldn't the suggested code, specifically this: 
```rust
fn get_thread_name(&self) -> &str {
    thread::current().name().unwrap_or("-")
}
```
create a temporary reference? So when its called in `create_prefix`, a `"returns a value referencing data owned by the current function"` error is thrown. 

Instead we added the following code:
```rust
fn get_thread_name(&self) -> String {
    thread::current().name().unwrap_or("-").to_string()
}
```
 Unfortunately, It still creates the String object. Any suggestions around temporary reference problem?

Reply:
I do not think we can get around without using a String here. What Sundar said is correct. We also use String objects for any part of the prefix,  storing them in a **Vec<String>**, before joining them into a bigger String at the end of the function. I believe it is okay to leave it as it is.